bridges/snowbridge/primitives/core/src/location.rs

```diff
@@ -230,4 +243,132 @@ mod tests {
 			);
 		}
 	}
+
+	#[test]
+	fn test_agent_id_fully_compatible() {
+		// Agent locations in V1 are not reanchored
+		let legacy_agent_locations = [
```

Comment:
We only need legacy compatibility for AssetHub itself `Location::new(1, [Parachain(1000)])`. All other locations can use the new format.

That is why I suggest you implement a custom `DescribeLocation` that does the following:
```
if location == AssetHub:
  DescribeV1(location)
else
  DescribeV2(location)
```

Reply:
We use a tuple for the backward compatibility.

https://github.com/yrong/polkadot-sdk/blob/381e71f85ec5ed5acac8fbbf0bc405496b1bb0fd/bridges/snowbridge/primitives/core/src/location.rs#L46 just for handling the legacy agent location like AH. 

I prefer the inheritance/combination approach over an if/else gate. More elegant, IMHO.

Reply:
Makes sense.